data/lavezzo2020suppression/lavezzo2020suppression-extraction.md

```diff
@@ -0,0 +1,250 @@
+        temp = temp.rename(columns = {'test_date':'first_'+ type})
+        temp['Round'] = round
+        temp_m = pd.concat([temp_m,temp],axis = 0,ignore_index=True)
+    df = df.merge(temp_m,how = 'left',on=['id','Round'])
+
+```
+
+```python
+#Create the PCR results
+df['pcr_results'] = 'Neg'
+df.loc[~df['RdRp'].isnull() | ~df['E'].isnull(),'pcr_results'] = 'Pos' 
+```
+
+```python
+#Assign the Test date
+df.loc[df['pcr_results'] == 'Pos','test_date'] = df['first_Pos']
+df.loc[df['pcr_results'] == 'Neg','test_date'] = df['first_Neg']
+
+  
+```
+
+```python
+#Retrieve the first positive date
+df_test_pos = df_test[df_test['results'] == 'Pos']
+df_test_pos = df_test_pos.groupby('id')['test_date'].agg(min).reset_index()
+df_test_pos = df_test_pos.rename(columns = {'test_date' : 'first_pos_date_overall'})
+df = df.merge(df_test_pos,on = 'id',how = 'left')
+```
+
+```python
+#Try to create the Reference date.
+df['reference_type'] = 'first_pos'
+df.loc[~df['first_symptoms_date'].isnull(),'reference_type'] = 'symptom'
+
+```
+
+```python
+#Create the numeric time
+df.loc[df['reference_type'] == 'first_pos','time'] = (df['test_date'] - df['first_pos_date_overall']).dt.days
+df.loc[df['reference_type'] == 'symptom','time'] = (df['test_date'] - df['first_symptoms_date']).dt.days
+
+```
+
+```python
+#Exclude missing time.
+df = df.loc[~df['time'].isnull(),]
+#Change the coding of negative pcr values.
+for i in ['RdRp','E']:
+    df.loc[df[i].isnull(),i] = 'negative'
+```
+
+```python
+#Enter the yaml writing stage
+participant_list = []
+for type in ['first_pos','symptom']:
+      df_output = df[df['reference_type'] == type]    
+      temp = [dict(attributes=dict(age=df_output.loc[df_output.loc[df_output["id"]==i].index[0],'age_group'].astype('object'),
+                                   sex=df_output.loc[df_output.loc[df_output["id"]==i].index[0],'gender']),
+                                   measurements=[dict(analyte='RdRp_'+type,
+                                                time=df_output.loc[(df_output['Round'] == j) & (df_output['id'] == i),"time"].item(),
+                                                value=df_output.loc[(df_output['Round'] == j) & (df_output['id'] == i),"RdRp"].item()) for j in np.unique(df_output.loc[df_output['id'] == i,'Round'])] +
+                                                [dict(analyte="E_"+type,
+                                                time=df_output.loc[(df_output['Round'] == j) & (df_output['id'] == i),"time"].item(),
+                                                value=df_output.loc[(df_output['Round'] == j) & (df_output['id'] == i),"E"].item()) for j in np.unique(df_output.loc[df_output['id'] == i,'Round'])]) for i in df_output['id']]
+      participant_list.extend(temp)
+
+
+```
+
+```python
+haisuproj = dict(title="Suppression of a SARS-CoV-2 outbreak in the Italian municipality of Vo’",
```

Comment:
```suggestion
haisuproj = dict(title="Suppression of a SARS-CoV-2 outbreak in the Italian municipality of Vo",
```

Maybe we can drop this last character here? It ends up being strangely in the yaml file.